src/base-api-client.ts

```diff
@@ -0,0 +1,90 @@
+import { ApiRequestHandler } from './api-request-handler';
+
+type ServiceApi = 'topic_extraction' | 'sentiment_analysis';
+
+/**
+ * Base client implementation. Intended to be extended by a specific client per API
+ */
+export class BaseApiClient<TJob, TResult> {
+    apiHandler: ApiRequestHandler;
+
+    /**
+     * @param accessToken Access token used to validate API requests
+     * @param serviceApi Type of api service
+     * @param version (optional) version of the API to be used
+     */
+    constructor (accessToken: string, serviceApi: ServiceApi, version = 'v1') {
```

Comment:
```suggestion
    constructor (accessToken: string, serviceApi: ServiceApi, version: string) {
```
We're the only ones who will be invoking this so I wonder if we should just make it explicit